sklearn/preprocessing/_encoders.py

```diff
@@ -544,7 +544,7 @@ def transform(self, X):
 
         indptr = np.empty(n_samples + 1, dtype=int)
         indptr[0] = 0
-        np.sum(X_mask, axis=1, out=indptr[1:])
+        np.sum(X_mask, axis=1, out=indptr[1:], dtype=indptr.dtype)
```

Comment:
This is a major fix for numpy 1.22 otherwise the encoder does not work at all. I think this should be backported to 1.0.2.